CliClient/tests/queryBuilder.js

```diff
@@ -0,0 +1,167 @@
+		});
+	});
+
+	it('when filter contains mulitple words in title', () => {
+		const testTitle = 'word1 word2'.split(' ');
+		const testBody = 'someBody';
+		const filters = new Map([
+			['body', [{ relation: 'AND', value: testBody }]],
+			[
+				'title',
+				[
+					{ relation: 'AND', value: testTitle[0] },
+					{ relation: 'AND', value: testTitle[1] },
+				],
+			],
+		]);
+
+		expect(queryBuilder(filters)).toEqual({
+			query: defaultMatchSQL,
+			params: [`title:${testTitle[0]} title:${testTitle[1]} body:${testBody}`],
+		});
+	});
+
+	it('when filter contains multiple words in body', () => {
+		const testTitle = 'testTitle';
+		const testBody = ['foo', 'bar'];
+		const filters = new Map([
+			['title', [{ relation: 'AND', value: testTitle }]],
+			[
+				'body',
+				[
+					{ relation: 'AND', value: testBody[0] },
+					{ relation: 'AND', value: testBody[1] },
+				],
+			],
+		]);
+
+		expect(queryBuilder(filters)).toEqual({
+			query: defaultMatchSQL,
+			params: [`title:${testTitle} body:${testBody[0]} body:${testBody[1]}`],
+		});
+	});
+
+	it('when filter contains title or body', () => {
+		const testTitle = 'testTitle';
+		const testBody = 'bodyTitle';
+		const filters = new Map([
+			['body', [{ relation: 'OR', value: testBody }]],
+			['title', [{ relation: 'AND', value: testTitle }]],
+		]);
+
+		expect(queryBuilder(filters)).toEqual({
+			query: defaultMatchSQL,
+			params: [`title:${testTitle} OR body:${testBody}`],
+		});
+	});
+
+	it('when filter contains text', () => {
+		const filters = new Map([
+			['text', [{ relation: 'AND', value: 'sampletext' }]],
+		]);
+		expect(queryBuilder(filters)).toEqual({
+			query: defaultMatchSQL,
+			params: ['sampletext'],
+		});
+	});
+
+	it('when filter contains multiple text words', () => {
+		const filters = new Map([
+			['text', [{ relation: 'AND', value: 'multiple words' }]],
+		]);
+		expect(queryBuilder(filters)).toEqual({
+			query: defaultMatchSQL,
+			params: ['multiple words'],
+		});
+	});
+
+	it('when filter contains phrase text words', () => {
+		const filters = new Map([
+			['text', [{ relation: 'AND', value: '"multiple words"' }]],
+		]);
+		expect(queryBuilder(filters)).toEqual({
+			query: defaultMatchSQL,
+			params: ['"multiple words"'],
+		});
+	});
+
+	it('when filter contains tags', () => {
+		let filters = new Map([
+			['tag', [{ relation: 'AND', value: 'tag123' }]],
+		]);
+		expect(queryBuilder(filters)).toEqual({
+			query: `WITH ${tagSQL} ${defaultSQL} AND ROWID IN (SELECT notes.ROWID from (tag_filter) JOIN notes on tag_filter.id=notes.id)`,
+			params: ['tag123'],
+		});
+
+		filters = new Map([
+			['tag', [{ relation: 'AND', value: 'tag123' }, { relation: 'AND', value: 'tag345' }]],
+		]);
+		expect(queryBuilder(filters)).toEqual({
+			query: `WITH ${tagDoubleSQL} ${defaultSQL} AND ROWID IN (SELECT notes.ROWID from (tag_filter) JOIN notes on tag_filter.id=notes.id)`,
+			params: ['tag123', 'tag345'],
+		});
+	});
+});
```

Comment:
I think it's not necessary to test the SQL strings as it's a bit too low-level, and might be difficult to maintain. Instead we should test the rows returned from these queries, which I believe you are already doing. So please could you take out this file off the PR for now?